ui/lib/apps/KeyViz/components/KeyViz.tsx

```diff
@@ -143,11 +147,15 @@ const KeyViz = () => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [])
 
-  const onChangeDateRange = useCallback((v: number) => {
-    setDateRange(v)
-    setSelection(null)
+  const onChangeDateRange = useCallback(
+    (v: number) => {
+      setDateRange(v)
+      setSelection(null)
+      resetAutoRefresh()
```

Comment:
I think you should revert these changes and only reset the timer in `updateHeatmap` or places like that, since it is the "request" behavior that causes a timer to be reset.